package com.ecommerce.project.service;

import com.ecommerce.project.payload.CartDTO;
import jakarta.transaction.Transactional;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.jpa.repository.Query;

import java.util.List;


public interface CartService {

     CartDTO addProductToCart(Long productId, Integer quantity) ;

    List<CartDTO> getAllCarts();

    CartDTO getCartByEmailCardId(String emailId, Long cartId);


    @Transactional
    CartDTO updateProductQuantityInCart(Long productId, Integer quantity);

    String deleteProductFromCart(Long cartId, Long productId);

    void updateProductInCart(Long cartId, Long productId);
}

